A47-xds-federation.md

```diff
@@ -0,0 +1,682 @@
+  }
+}
+```
+
+The new `client_default_listener_resource_name_template` field will have
+the following format:
+
+```
+// A template for the name of the Listener resource to subscribe to
+// for a gRPC client channel.  Used only when the channel is created
+// with an "xds:" URI with no authority.
+//
+// If starts with "xdstp:", will be interpreted as a new-style name,
+// in which case the authority of the URI will be used to select the
+// relevant configuration in the "authorities" map.
+//
+// The token "%s", if present in this string, will be replaced with
+// the service authority (i.e., the path part of the target URI
+// used to create the gRPC channel).  If the template starts with
+// "xdstp:", the replaced string will be percent-encoded.
+//
+// Defaults to "%s".
+"client_default_listener_resource_name_template": <string>,
+```
+
+There will be two changes to the existing
+`server_listener_resource_name_template` described in [gRFC A36][A36]:
+- If the template starts with `xdstp:`, it will be interpreted as a
+  new-style name, in which case the authority of the URI will be used to
+  select the relevant configuration in the `authorities` map.
+- When replacing the `%s` token, if the template starts with `xdstp:`,
+  the replacement string will be percent-encoded in the resulting resource
+  name.
+
+The resulting field will have the following format:
+
+```
+// A template for the name of the Listener resource to subscribe to
+// for a gRPC server.
+//
+// If starts with "xdstp:", will be interpreted as a new-style name,
+// in which case the authority of the URI will be used to select the
+// relevant configuration in the "authorities" map.
+//
+// The token "%s", if present in this string, will be replaced with
+// the IP and port on which the server is listening.  If the template
+// starts with "xdstp:", the replaced string will be percent-encoded.
+//
+// There is no default; if unset, xDS-based server creation fails.
+//
+// Example:
+// "xdstp://xds.example.com/envoy.config.listener.v3.Listener/grpc/server/%s"
+"server_listener_resource_name_template": <string>,
+```
+
+#### Example: Bootstrap Config Contains No New Fields
+
+If the bootstrap config does not contain any of the new fields described
+above, then here's how the resource names will be determined.
+
+If a gRPC client channel is created for `xds:server.example.com`:
+- The target URI has no authority, so gRPC will look at the bootstrap
+  config's `client_default_listener_resource_name_template` field, which
+  is not set, so it defaults to `%s`.  So the Listener resource name will be
+  `server.example.com`, just like it would be prior to this design.
+- The Listener resource name does not start with `xdstp:`, so we do not
+  use the `authorities` map at all.
+- We use the top-level `xds_server` list, which is the same one that we
+  would have used prior to this design.
+
+If a gRPC client channel is created for `xds://xds.example.com/server.example.com`:
```

Comment:
Having "example.com" as the top-level-domain in both the authority and the path part has no significance other than it being a typical use-case, is that right? So this example could very well be `xds://xds.authority.com/server.example.com` . If true, wouldn't it be better to not use "example.com" in both places to avoid implying some sort of connection?

Reply:
Done.